packages/visx-shape/src/shapes/Pie.tsx

```diff
@@ -78,14 +83,28 @@ export default function Pie<Datum>({
   // eslint-disable-next-line react/jsx-no-useless-fragment
   if (children) return <>{children({ arcs, path, pie })}</>;
 
+  const fillGenerator = getFillGenerator(fill);
   return (
     <Group className="visx-pie-arcs-group" top={top} left={left}>
       {arcs.map((arc, i) => (
         <g key={`pie-arc-${i}`}>
-          <path className={cx('visx-pie-arc', className)} d={path(arc) || ''} {...restProps} />
+          <path
+            className={cx('visx-pie-arc', className)}
+            d={path(arc) || ''}
+            fill={fillGenerator(arc)}
+            {...restProps}
+          />
           {centroid?.(path.centroid(arc), arc)}
         </g>
       ))}
     </Group>
   );
 }
+
+function getFillGenerator<Datum>(fill: string | StringAccessor<Datum>): StringAccessor<Datum> {
```

Comment:
Not sure where this should go so I threw it at the bottom. Please let me know if there is a better place for this.

Reply:
See my comment above about potentially removing this/simplifying the logic, but if you'd like to keep it I think inside the file is fine :)